pandas/core/arrays/masked.py

```diff
@@ -229,6 +231,30 @@ def to_numpy(
             data = self._data.astype(dtype, copy=copy)
         return data
 
+    def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
+        dtype = pandas_dtype(dtype)
+
+        if is_dtype_equal(dtype, self.dtype):
+            if copy:
+                return self.copy()
+            return self
+
+        # if we are astyping to another nullable masked dtype, we can fastpath
+        if isinstance(dtype, BaseMaskedDtype):
+            # TODO deal with NaNs for FloatingArray case
+            data = self._data.astype(dtype.numpy_dtype, copy=copy)
+            # mask is copied depending on whether the data was copied, and
+            # not directly depending on the `copy` keyword
+            mask = self._mask if data is self._data else self._mask.copy()
+            cls = dtype.construct_array_type()
+            return cls(data, mask, copy=False)
+
+        if isinstance(dtype, ExtensionDtype):
+            cls = dtype.construct_array_type()
+            return cls._from_sequence(self, dtype=dtype, copy=copy)
+
+        return super().astype(dtype, copy=copy)
```

Comment:
this is in theory not reachable?

Reply:
not with the existing subclasses.  this still seemed like the idiomatic thing to do

Reply:
I would personally remove it then, if it's not reachable at the moment. 
Then you can also remove the `if isinstance(dtype, ExtensionDtype):` from the previous block.

Reply:
im not wild about this method relying on the current behavior of the existing subclasses.  better to conform to the standard patterns, even if it means a couple of extra checks that turn out to be unnecessary

Reply:
Could also call it `_astype` instead of calling `super()`, to make it clearer it is a shared helper method, and not an actual (full) base implementation